pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -948,16 +943,16 @@ public String toString(OffsetTime offsetTime) {
 
   public String toString(OffsetDateTime offsetDateTime) {
     try (ResourceLock ignore = lock.obtain()) {
-      if (offsetDateTime.isAfter(MAX_OFFSET_DATETIME)) {
+      if (offsetDateTime.equals(OffsetDateTime.MAX)) {
         return "infinity";
-      } else if (offsetDateTime.isBefore(MIN_OFFSET_DATETIME)) {
+      } else if (offsetDateTime.equals(OffsetDateTime.MIN)) {
         return "-infinity";
       }
 
```

Comment:
I don't think that +- infinity actually has a value
from pg docs
```
infinity | date, timestamp | later than all other time stamps
```
So I don't think we can say that if it equal to OffsetDateTime.MIN then it is infinity.

Reply:
```java
    /**
     * The minimum supported {@code OffsetDateTime}, '-999999999-01-01T00:00:00+18:00'.
     * This is the local date-time of midnight at the start of the minimum date
     * in the maximum offset (larger offsets are earlier on the time-line).
     * This combines {@link LocalDateTime#MIN} and {@link ZoneOffset#MAX}.
     * This could be used by an application as a "far past" date-time.
     */
    public static final OffsetDateTime MIN = LocalDateTime.MIN.atOffset(ZoneOffset.MAX);
```

>This could be used by an application as a "far future" date-time.

I think it is quite close to `-infinity`.

What are your suggestions?

Reply:
Essentially postgres is treating it mathematically as infinity so I don't think there are any actual dates or times that map to infinity as small or as large as they might be. Here's one idea https://github.com/npgsql/npgsql/blob/ea3932d8de0e886a5c3783657d498071e33ac1e1/src/Npgsql/Internal/TypeHandlers/DateTimeHandlers/TimestampTzHandler.cs#L54 They have a flag to disable and throw an error if they receive `infinity` as a value , otherwise they chose values which map to infinity. So as I see it we are choosing to not support the values that we are mapping to infinity so that we can support `infinity`. I think their solution is a better path. We allow the user to chose between not supporting `infinity` or not.

Reply:
MIN/MAX was mapped to infinity for quite some time in pgjdbc, so we can't rip it off.
However, we should not add more properties to this PR.

I'm not sure there's a demand for "throw exceptions on infinity values" though.

Reply:
upon reflection I agree. Just map it to a value. Apparently it is mapped to a value in the database so we aren't consuming a legitimate value